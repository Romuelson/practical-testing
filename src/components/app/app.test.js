import { render, screen, fireEvent } from '@testing-library/react';
import App from './app';

describe('Компонент: App', () => {
	it('should-0', () => {
		render(<App />);

		const appTitle = screen.getByText(/hello world/i);
		const appButton = screen.getByRole('button');
		const appInput = screen.getByPlaceholderText(/input value.../i);

		expect(appTitle).toBeInTheDocument();
		expect(appButton).toBeInTheDocument();
		expect(appInput).toBeInTheDocument();

		expect(appInput).toMatchSnapshot();

		// screen.debug();
	});

	it('should-1', async () => {
		render(<App />);

		// const appTitle = screen.quertByText(/hello world2/i);
		// expect(appTitle).toBeNull();

		const appTitle = await screen.findByText(/data/i);
		expect(appTitle).toBeInTheDocument();
		expect(appTitle).toHaveStyle({ color: 'red' });
	});

	it('should-2', () => {
		render(<App />);

		const appInput = screen.getByPlaceholderText(/input value.../i);
		const toggleBtn = screen.getByTestId('toggleBtn');

		expect(screen.queryByTestId('valueElm')).toContainHTML('');
		expect(screen.queryByTestId('toggleElm')).toBeNull();

		fireEvent.input(appInput, {
			target: {value: '123123'}
		})

		// userEvent.type(appInput, '123123');
		fireEvent.click(toggleBtn);

		expect(screen.queryByTestId('valueElm')).toContainHTML('123123');
		expect(screen.getByTestId('toggleElm')).toBeInTheDocument();
	});
});
