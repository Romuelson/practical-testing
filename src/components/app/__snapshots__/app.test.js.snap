// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент: App should-0 1`] = `
<input
  class="app_input"
  placeholder="input value..."
  type="text"
/>
`;
